bin/specextract

```diff
@@ -79,381 +81,449 @@ v4 = make_verbose_level(toolname, 4)
+    with new_pfiles_environment(ardlib=False,copyuser=False):
+        # ARF output file
+        carffile = f"{full_outroot}.corr.arf"
+
+        arfcorr.punlearn()
+        arfcorr.infile = reg_image
+        arfcorr.arf = orig_arf
+        arfcorr.outfile = carffile
+        arfcorr.region = spec.get_region_filter(infile)[1]
+        arfcorr.x = skyx
+        arfcorr.y = skyy
+        arfcorr.energy = 0
+        arfcorr.verbose = 0
+        arfcorr.clobber = "yes"
+
+        arfcorr()
+
+        # add arfcorr history to file, since it doesn't automatically
+        param_arfcorr = {}
+        for acpar in arfcorr._parnames:
+            param_arfcorr[acpar] = arfcorr._get_param_value(acpar)
+
+        add_tool_history(carffile,"arfcorr",param_arfcorr)
+    
+    return carffile
 
-                    file_status.append(inf)
 
-    if file_status != []:
-        raise IOError("Input event file(s) missing necessary Repro IV header keywords.  Reprocess data with chandra_repro or add the keywords to the event file(s) with r4_header_update.")
 
+def determine_rmf_tool(infile,rmffile,verbose):
+    """
+    Decide whether to run mkrmf or mkacisrmf, return the name of the tool to run.
+    """
+    
+    caldb_p2_resp_file = ""   # P2_RESP CALDB file
 
-##########################################################################
-# Usage:
-#  extract_spectra( full_outroot, infile, ptype, ewmap,
-#                   binwmap, instrument, clobber, verbose)
-#
-# Aim:
-#   Create spectrum from input file.
-#
-##########################################################################
+    # NB: Adjust call_calquiz() function eventually
+    #     to avoid this step of having to initialize
+    #     the variable.
 
-def extract_spectra(full_outroot, infile, ptype, channel, ewmap, binwmap, instrument, clobber, verbose):
-    # output spectrum filename
-    specfile = f"{full_outroot}.{ptype.lower()}"
+    # rmftool               # rmf tool to use: 'mkrmf' or 'mkacisrmf'
+    # numCalFiles           # number of CALDB files found by calSearch
 
-    dmextract.punlearn()
+    # query the CALDB for the p2_resp file
 
-    dmextract.outfile = specfile
-    dmextract.opt = "pha1"
-    dmextract.clobber = clobber
-    dmextract.verbose = str(verbose)
+    v2("Searching for P2_RESP calibration file...")
 
-    if channel == "1:1024:1":
-        dmextract.infile = f"{infile}[bin {ptype}]"
-    else:
-        dmextract.infile = f"{infile}[bin {ptype}={channel}]"
+    calquiz_results = call_calquiz(infile,"SC_MATRIX",rmffile,caldb_p2_resp_file,verbose)
+    numCalFiles = calquiz_results[0]
 
-    if instrument == "ACIS":
-        dmextract.wmap = f"[energy={ewmap}][bin {binwmap}]"
-    else:
-        dmextract.wmap = ""
+    caldb_p2_resp_file = calquiz_results[1]
 
-    dmextract()
+    v3(f"Found the following P2_RESP file(s): {caldb_p2_resp_file}")
 
-    return specfile
+    if not numCalFiles or numCalFiles in [0.0, 0, 0.]:
```

Comment:
I haven't got to `call_calquiz` yet, but in general I would expect that the return value should be an integer or, even better, you just return a list of names so you can use `len` to identify how many files are matched. If you want to return more-structured data than return something like a dictionary or a namedtuple.